### ISSUE TITLE
c8run release artifacts and testing process should be more intertwined with zeebe build process

### ISSUE THREAD

**Issue by jessesimpson36** [camunda/camunda#23586]
Related to https://github.com/camunda/engineering-leadership-team/issues/256

We do not want the c8run distribution to be a separate thing from the camunda core, and regarding the release-process, releasing the camunda core should include what's necessary to run it using c8run.

This means:
1. There should no longer be a [c8run github release](https://github.com/camunda/camunda/releases/tag/c8run-8.6.2) and in it's place, the artifacts should be uploaded to the [camunda core release](https://github.com/camunda/camunda/releases/tag/8.6.3) .
2. The c8run scripts should not download a release of zeebe, and should instead use the build of zeebe generated from a CI process.
3. The c8run tests should run generally whenever Camunda Core source code is modified.  We may need to revisit the artifact retention policy for the builds of c8run, since those are quite large, and there are a lot of CI builds in this repo.

**Comment by jessesimpson36** [comment]
There are some problems trying to introduce c8run into the build process of camunda as a whole.  the builds take a long time when we do it this way, and we still depend on connectors so if we had a build for a camunda release, connectors may release their new image and then it would invalidate the c8run build.

I think it's still valuable to be able to build c8run from source and make that process easier, but just not part of the ci process.

For the release process, we can upload our builds to the github release for camunda and not a separate release artifact.  And so that we know the status of this process, we will track 
https://github.com/camunda/Release-Train/issues/15 to get c8run included into the release train.

**Comment by MaxTru** [comment]
@jessesimpson36 @vheinila 
we state that...

> For the release process, we can upload our builds to the github release for camunda and not a separate release artifact

...so that I understand correctly: we will not have a separate release in camunda/camunda (not like here: https://github.com/camunda/camunda/releases/tag/c8run-8.6.2), but we will add the assets to the C8 release (e.g., add here: https://github.com/camunda/camunda/releases/tag/8.6.4).
@jessesimpson36 : will these be a manual step carried out be the distro release manager? Should this be a separate step in the release train (to ensure and remind this is conducted)?

**Comment by jessesimpson36** [comment]
>  we will not have a separate release in camunda/camunda (not like here: https://github.com/camunda/camunda/releases/tag/c8run-8.6.2), but we will add the assets to the C8 release (e.g., add here: https://github.com/camunda/camunda/releases/tag/8.6.4).

That's correct. I think the separate c8run release caused some confusion when some people were looking for the artifacts. 

> will these be a manual step carried out be the distro release manager?

For now, this will be a manual step.  all that needs done is to download the release build from the github actions workflow and upload it to the release page. Not hard, but it would help for this step to be part of the release train. In the future, having this be automated would be nice.  I think at this point, Ville has already added the artifact build/release to the release train. though I'm not sure if that change made it in for the 8.6.4 release train. I will check on that.

**Comment by hisImminence** [comment]
Hey @jessesimpson36,

IMO we can close this one and follow up on improvements moving forward. C8Run is now a dedicated process in the release process and separate from the camunda/camudna build. Please reopen or create a more specific follow up if necessary ;)
